lib/elixir/lib/calendar/datetime.ex

```diff
@@ -437,6 +437,29 @@ defmodule DateTime do
   end
 
   @doc """
+  Parses the extended "DateTimes" format described by
+  [ISO 8601:2004](https://en.wikipedia.org/wiki/ISO_8601).
+
+  Raises if the format is invalid.
+
+  ## Examples
+
+      iex> DateTime.from_iso8601!("2015-01-23T00:00:00.000000Z")
+      #DateTime<2015-01-23 00:00:00.000000Z>
```

Comment:
I think we should not drop `offset` information, right?

Reply:
That's a good point, it is probably the reason why we don't have the bang variant. I don't think we should automatically drop information, the users should do it explicitly. So pattern matching to `from_iso8601` may arguably be better.

Reply:
Why does `iso_8601/2` not add the offset information to DateTime's `utc_offset`?

Reply:
For the reasons I said in the mailing list. :) Here is what the docs say as well: https://github.com/anthonator/elixir/blob/c31b05b8dc1fdefa0e5c8b03c35b9a06c626a4a8/lib/elixir/lib/calendar/datetime.ex#L352-L355

I thought we didn't have from_iso8601 at all because of the above but returning the offset also makes sense.

Reply:
After reading up some more on ISO8601 I think what you guys are saying makes sense. I'll close this out.

Reply:
my first reaction was against it for the reasons above, but maybe it's still worth special-casing UTC datetime (which isn't trivial, as `"Z"` parsing is deep in `Calendar.ISO`)? So `from_iso8601!("2015-01-23T00:00:00Z")` would return the datetime because it's always unambiguous (offset=0, std_offset=0, etc), but e.g.: `from_iso8601!("2015-01-23T00:00:00+00:00")` and others raise an error?